libbeat/kibana/index.go

```diff
@@ -0,0 +1,121 @@
+package kibana
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type Index struct {
+	Version   string
+	IndexName string
+	BeatDir   string
+	BeatName  string
+
+	fieldsYaml       string
+	targetDirDefault string
+	targetDir5x      string
+	targetFilename   string
+}
+
+// Create the Index-Pattern for Kibana for 5.x and default.
+func (i *Index) Create() ([]string, error) {
+	indices := []string{}
+
+	err := i.init()
+	if err != nil {
+		return nil, err
+	}
+
+	commonFields, err := common.LoadFieldsYaml(i.fieldsYaml)
+	if err != nil {
+		return nil, err
+	}
+	transformed := TransformFields("@timestamp", i.IndexName, commonFields)
+
+	if fieldsBytes, err := json.Marshal(transformed["fields"]); err != nil {
+		return nil, err
+	} else {
```

Comment:
I assume the else here is needed because you only assign `fieldsBytes` above. My personal preference would be not have the else but make the call outside the if and then only check `if err != nil { ` 

Same below.